core/src/packet.rs

```diff
@@ -32,37 +36,64 @@ impl<'a, U> Packet<'a, U>
 where
     U: UserDataPtr,
 {
-    pub fn submit(mut self) {
+    pub fn new(handle: ClientHandle<'a, U>, user_data: U, operation: Operation) -> Self {
+        let user_data = U::into_raw_const_ptr(user_data)
+            .cast::<std::ffi::c_void>()
+            .cast_mut();
+
+        let raw = Box::new(sys::tb_packet_t {
+            next: ptr::null_mut(),
+            user_data,
+            operation: operation.0,
+            status: 0,
+            data_size: 0,
+            data: ptr::null_mut(),
+            batch_next: ptr::null_mut(),
+            batch_tail: ptr::null_mut(),
+            batch_size: 0,
+            reserved: [0; 8],
+        });
+
+        let ptr = ManuallyDrop::new(Box::into_raw(raw));
+
+        Self { raw: ptr, handle }
+    }
+
+    pub fn submit(mut self) -> Result<(), ClientClosedError> {
```

Comment:
@jeffutter I don't really understand this change. Judging from the body, it doesn't return this error anywhere. So, why have it in the signature?